Index: django/contrib/admin/options.py
===================================================================
--- django/contrib/admin/options.py	(revision 13431)
+++ django/contrib/admin/options.py	(working copy)
@@ -798,6 +798,8 @@
                     self.save_formset(request, form, formset, change=False)
 
                 self.log_addition(request, new_object)
+                if hasattr(self, "post_save"):
+                    self.post_save(new_object)
                 return self.response_add(request, new_object)
         else:
             # Prepare the dict of initial data from the request.
@@ -899,6 +901,8 @@
 
                 change_message = self.construct_change_message(request, form, formsets)
                 self.log_change(request, new_object, change_message)
+                if hasattr(self, "post_save"):
+                    self.post_save(new_object)
                 return self.response_change(request, new_object)
 
         else:
